main.py

```diff
@@ -7,41 +17,66 @@ def get_arguments(arguments):
     number_of_arguments = len(arguments)
     print("Number of arguments", number_of_arguments)
 
-    if number_of_arguments <= 1 or number_of_arguments > 2:
+    if number_of_arguments < 1 or number_of_arguments > 2:
         print("Incorrect command usage, please follow the correct format")
         print("COMMAND: python3 [link] [max_repos]")
-        return []
+        return command_failed()
     return arguments
 
 
 def validate_arguments(arguments):
-    if len(arguments) == 0:
-        return
+    if arguments is None:
+        return command_failed()
 
+    first_argument = arguments[0]
     second_argument = arguments[1]
 
+    try:
+        if first_argument.startswith("http"):
+            pass
+        else:
+            return command_failed()
+    except ValueError:
+        print("Incorrect command usage, argument 1 is not a valid URL.")
+        return command_failed()
+
     try:
         if int(second_argument):
             pass
     except ValueError:
         print("Incorrect command usage, argument 2 is not an integer")
+        return command_failed()
+
+    return arguments
 
 
 def generate_url(arguments):
+    if arguments is None:
+        return command_failed()
+
```

Comment:
If you exit in command_failed, arguments is guaranteed to not be None here because you are already checking in the `validate_arguments` function, so you can remove this

Reply:
Good thinking, implemented here [b0774e3].